src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -441,6 +453,10 @@ public AmazonCorrettoCryptoProvider() {
     this.shouldRegisterSecureRandom =
         Utils.getBooleanProperty(PROPERTY_REGISTER_SECURE_RANDOM, true);
 
+    // Register EdDSA if JDK version >= 15
+    String[] version = System.getProperty("java.version").split("\\.");
```

Comment:
we have a [util](https://github.com/corretto/amazon-corretto-crypto-provider/blob/cbe344117f5a2bcdd6777100857c506cbc2a4904/src/com/amazon/corretto/crypto/provider/Utils.java#L535) for this